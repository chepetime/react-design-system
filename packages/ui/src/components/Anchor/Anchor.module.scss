@use "./../../styles" as styles;

:root {
  --anchor-background-color: #{styles.$pink-500};
}

:global .dark {
  --anchor-background-color: #{styles.$indigo-500};
}

.Anchor {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--anchor-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
