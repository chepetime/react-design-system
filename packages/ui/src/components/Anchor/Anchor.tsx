import React from "react";
import classnames from "classnames";
import css from "./Anchor.module.scss";

export interface AnchorProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Anchor = React.forwardRef<HTMLDivElement, AnchorProps>(
  (props: AnchorProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Anchor)} ref={ref}>
        <p>Anchor component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Anchor.displayName = "Anchor";
