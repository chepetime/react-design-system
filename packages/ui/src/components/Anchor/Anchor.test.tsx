
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Anchor } from "./Anchor";
import { render, screen } from "@testing-library/react";

describe("Anchor", () => {
  it("renders without errors", () => {
    render(<Anchor />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Anchor>Some children</Anchor>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Anchor>Some children</Anchor>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Anchor />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
