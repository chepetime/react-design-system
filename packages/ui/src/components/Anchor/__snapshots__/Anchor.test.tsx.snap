// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor matches snapshot 1`] = `
<div>
  <div
    class="Anchor"
  >
    <p>
      Anchor component working!
    </p>
    <div />
  </div>
</div>
`;
